docs/pipelines/overview/configuration.md

```diff
@@ -1,95 +1,368 @@
+
+The expected schema for the `accounts.yml` file is as follows:
+
+```yaml
+# required: Name of an account
+an_account:
+  # required: The AWS account ID
+  id: "an-aws-account-id"
+  # optional: The email address of the account owner
+  owner_email: "an-email-address"
+  # optional: Whether or not a VPC has been created in the account. Default is false.
+  vpc_created: true
+```
+
+Note that multiple AWS Accounts blocks can be defined, pointing to different `accounts.yml` files. This allows for the segmentation of AWS accounts into different YAML files for organizational purposes.
+
+The decision to leverage YAML files instead of HCL files for defining the configurations for AWS accounts was an intentional decision to increase the portability of these configurations for usage outside of Pipelines. Tools like [Terragrunt](https://github.com/gruntwork-io/terragrunt/) and [yq](https://github.com/mikefarah/yq) can be used to leverage these files, as they are more portable than HCL files.
+
+*Supported Attributes:*
+
+- `path` (Required): The path to the YAML file that contains the definitions of AWS accounts.
+
+### Pipelines Blocks
+
+Pipelines blocks are used to define configurations that are applicable to Pipelines itself.
+
+e.g.
+
+```hcl
+pipelines {
+  deploy_branch_name = "main"
+}
+```
+
+In this example, the `deploy_branch_name` attribute is set to `main`, which means that Pipelines will deploy infrastructure changes when the `main` branch is updated.
+
+*Supported Attributes:*
+
+- `deploy_branch_name` (Optional): The branch that Pipelines will deploy infrastructure changes from. If not set, Pipelines will deploy infrastructure changes from the `main` branch.
+
+### Annotation Blocks
+
+Annotation blocks are used to define configurations that are applicable to multiple units within a repository. These configurations are typically used to provide metadata about configurations that can be used by Pipelines to make decisions about how to interact with those configurations.
+
+The label applied to an annotation block is the name of the annotation. This is a user-defined label for the annotation, and must be globally unique.
+
+e.g.
+
+```hcl
+annotation "dev" {
+  filter {
+    paths = ["*dev/*"]
+  }
+
+  labels = {
+    "environment" = "dev"
+  }
+}
+```
+
+In this example, the `an_annotation` annotation is defined to match all units located within a directory that contains `dev` at the end of the path. All units that match this filter will have the label `key` set to `value`.
+
+*Supported Blocks:*
+
+- `filter` (Required): A filter block that determines which units the annotation is applicable to. See [Filter Blocks](#filter-blocks) for more information.
+- `labels` (Required): A map of key-value pairs that provide metadata about the units that the annotation is applicable to.
```

Comment:
Might be worthwhile pointing out that labels aren't used by pipelines today, but will be in the future